src/main/java/games/strategy/triplea/delegate/BattleDelegate.java

```diff
@@ -571,8 +571,10 @@ private static void setupTerritoriesAbandonedToTheEnemy(final BattleTracker batt
       return;
     }
     final PlayerID player = bridge.getPlayerID();
-    final Iterator<Territory> battleTerritories = Match.getMatches(data.getMap().getTerritories(), Matches
-        .territoryHasEnemyUnitsThatCanCaptureTerritoryAndTerritoryOwnedByTheirEnemyAndIsNotUnownedWater(player, data))
+    final Iterator<Territory> battleTerritories = Match
+        .getMatches(
+            data.getMap().getTerritories(),
+            Matches.territoryHasEnemyUnitsThatCanCaptureItAndIsOwnedByTheirEnemyAndIsNotUnownedWater(player, data))
```

Comment:
This is way too long of a method name!
As this has only one usage: What about splitting this into multiple matches and grouping them together with `Match#allOf`?

Reply:
There's no way I'm going to attempt to refactor this method without having characterization tests in place. :smile:

We really don't have any precedent for writing tests for `Match` implementations, which is probably not a good thing.  I've been meaning for a while to see how hard it would be to write some `Match` tests.  So maybe this would be a good opportunity to do that.

I'll take a stab at it, but if it starts getting extremely painful, I'm just going to punt and say the bang-for-buck value is too low to justify to fix a single Checkstyle violation.